core/runtime/impl/runtime_api.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_RUNTIME_RUNTIME_API_HPP
+#define KAGOME_CORE_RUNTIME_RUNTIME_API_HPP
+
+#include <utility>
+
+#include "common/buffer.hpp"
+#include "runtime/impl/wasm_executor.hpp"
+#include "runtime/wasm_memory.hpp"
+#include "runtime/wasm_result.hpp"
+#include "scale/scale.hpp"
+
+namespace kagome::runtime {
+  /**
+   * @brief base class for all runtime apis
+   */
+  class RuntimeApi {
+   public:
+    RuntimeApi(common::Buffer state_code,
```

Comment:
probably constructor can be protected

Reply:
It will be hard to test runtime API in this case